src/launcher.c

```diff
@@ -414,6 +459,11 @@ static int load_cxlib(char *cxlib_args)
     return -1;
   }
 
+  if (mprotect(p, sh_size, PROT_READ | PROT_EXEC) != 0) {
+    warn("mprotect(PROT_READ | PROT_EXEC)");
```

Comment:
Maybe be more explicit in the warning or add a comment about what this code does